tools/iree-e2e-matmul-test.c

```diff
@@ -224,6 +228,67 @@ static void write_int_element(iree_hal_element_type_t element_type, int value,
  * particular test program is entrenched here.
  *
  *****************************************************************************/
+// Write an int32_t element to a mapped row-major matrix buffer.
+static void write_int_to_matrix_element(int32_t value, iree_hal_dim_t m_size,
+                                        iree_hal_dim_t n_size,
+                                        iree_hal_element_type_t result_type,
+                                        void* data, iree_hal_dim_t m,
+                                        iree_hal_dim_t n) {
+  iree_host_size_t index = n + m * n_size;
+  (void)m_size;
+  if (iree_hal_element_type_is_integer(result_type, 32)) {
+    ((int32_t*)data)[index] = value;
+    return;
+  } else if (result_type == IREE_HAL_ELEMENT_TYPE_FLOAT_16) {
+    ((uint16_t*)data)[index] = iree_math_f32_to_f16((float)value);
+    return;
+  } else if (result_type == IREE_HAL_ELEMENT_TYPE_FLOAT_32) {
+    ((float*)data)[index] = value;
+    return;
+  }
+  iree_status_abort(iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
```

Comment:
you can use IREE_CHECK_OK here instead of directly calling abort - that helps debugging as the status is put onto the stack and will be available in a debugger

Reply:
Done. Thanks for checking it. 

Is the guidance here is to use `IREE_CHECK_OK` just here or everywhere else too?